src/utils/pinataUpload.ts

```diff
@@ -0,0 +1,58 @@
+import { PinataSDK } from "pinata";
+import toast from "react-hot-toast";
+
+// Interface for Pinata upload response
+interface PinResponse {
```

Comment:
very good to type but I'm surprised that this type is not directly present into the sdk / type-sdk